### DISCUSSION TITLE
What version of Guzzle does Laravel 8 require?

### DISCUSSION THREAD

**Discussion by elijahchancey** [laravel/framework#34949] [Q&A] [answered]
The Laravel 8 upgrade guide states Guzzle 7 is required:
https://github.com/laravel/docs/blob/8.x/upgrade.md

Composer.json in laravel/framework states Guzzle 6 is acceptable:
https://github.com/laravel/framework/blob/f6783a510d1380abca0ffec2eecdc126f69dc0fe/composer.json#L84

Thanks for clarifying!

**Comment by GrahamCampbell** [comment]
That's just because the default laravel/laravel app now requires that version of Guzzle, and we want to encourage everyone to migrate to Guzzle 7., Guzzle 6 is no longer receiving bug fixes.

NB Those require-dev lines are not the correct place to look to answer the question "what version of X does laravel support". The require and suggest entries are the correct place.

**Comment by slackerzz** [comment]
Maybe you can refer to

https://github.com/laravel/laravel/blob/master/composer.json#L14

**Reply by GrahamCampbell** [reply]
No, that is just the default. The authoritative requirements are listed in the suggest block of the framework's composer.json, as is the convention for all composer packages. Composer will even tell you what version to install if you don't have guzzle installed, if you read the output from `composer update`.

**Comment by GrahamCampbell** [comment, marked as answer]
The requirements are as follows:

* Laravel 5.8: Guzzle `^6.0` (https://github.com/laravel/framework/blob/v5.8.38/composer.json#L124)
* Laravel 6: Guzzle `^6.3.1|^7.0.1`(https://github.com/laravel/framework/blob/v6.20.2/composer.json#L126)
* Laravel 7: Guzzle `^6.3.1|^7.0.1` (https://github.com/laravel/framework/blob/v7.29.2/composer.json#L132)
* Laravel 8: Guzzle `^6.5.5|^7.0.1` (https://github.com/laravel/framework/blob/v8.12.3/composer.json#L135)

The upgrading guide simply describes the recommended upgrading path, since Guzzle 6 is now end of life, and will be dropped from Laravel in the future (probably in Laravel 9 or 10).
